### DISCUSSION TITLE
View Image node using base64

### DISCUSSION THREAD

**Discussion by WASasquatch** [comfyanonymous/ComfyUI#153] [Ideas]
A node to display images you don't want to save, display a base64 inline version.

**Comment by comfyanonymous** [comment]
There is an image preview node. The reason it doesn't use base64 is because image files can be quite big.

**Reply by WASasquatch** [reply]
Shouldn't be an issue. Colab have used base64 for ever now and no one has issues. Largest files anyone's gonna be getting is like 2.5 - 5mb (with up to 24gb VRAM)

Writing things to hard drive people don't intend to save when they ARE big is the issue. Comfy could be on a different HDD then browser, etc.

**Reply by WASasquatch** [reply]
Another issue is I am using Comfy as a node network and editing images, my current workspace, to make sure all post processing is correct is viewing 6 images before the final I intend to save. That 7 images being saved to disk every run.

**Reply by comfyanonymous** [reply]
Since you can see the history in ComfyUI the images on the PreviewImage node need to be put somewhere and it's better to put them in a temp directory on the disk that gets cleared than to keep them in memory.

The reasoning for not using base64 is that base64 adds 33% more data so it would be slower especially on places like colab where the bandwidth is sometimes slow because of localtunnel/etc...

**Reply by WASasquatch** [reply]
> Since you can see the history in ComfyUI the images on the PreviewImage node need to be put somewhere and it's better to put them in a temp directory on the disk that gets cleared than to keep them in memory.
> 
> The reasoning for not using base64 is that base64 adds 33% more data so it would be slower especially on places like colab where the bandwidth is sometimes slow because of localtunnel/etc...

But yet, you can literally only get images on colab via base64, there is no storage access to the output frame. Every one is viewing it this way, and no one really has issues. Disco Diffusion, Stable Diffusion notebooks like NOPs or my own Easy Diffusion, etc, all output base64. Base64 should be stored with the temp in the browser, only loaded into memory for it's rendering briefly, which is the same for a stored image on disk as well. 

Also isn't standard temp limits only 100 MB or so before being flushed? The browsers temp for webpages is far bigger.

**Reply by edankwan** [reply]
It would be great to have an option to use base64 instead. I want to run a local passive interface with auto query and I am worried it will wear off the SSD.

**Reply by ltdrdata** [reply]
> It would be great to have an option to use base64 instead. I want to run a local passive interface with auto query and I am worried it will wear off the SSD.

You'd better use websocket version.
![image](https://github.com/user-attachments/assets/155ad1d0-2b6e-4544-8698-533bd9936f6a)
